From b5fbfaf52f0f480e1c6ba8291f1ae69ce3c8f3fd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B8rn=20Damstedt=20Rasmussen?= <contact@bearware.dk>
Date: Thu, 5 Oct 2023 20:14:02 +0200
Subject: [PATCH] Fix 'pulseaudioHostApi' use-after-free in
 PaPulseAudio_Initialize

The call to PaPulseAudio_UnLock( pulseaudioHostApi->mainloop ) in
error-label is performed on 'pulseaudioHostApi' after
'pulseaudioHostApi' has been freed by PaPulseAudio_Free
---
 src/hostapi/pulseaudio/pa_linux_pulseaudio.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/hostapi/pulseaudio/pa_linux_pulseaudio.c b/src/hostapi/pulseaudio/pa_linux_pulseaudio.c
index 27dcdf8..e784b65 100644
--- a/src/hostapi/pulseaudio/pa_linux_pulseaudio.c
+++ b/src/hostapi/pulseaudio/pa_linux_pulseaudio.c
@@ -758,7 +758,6 @@ PaError PaPulseAudio_Initialize( PaUtilHostApiRepresentation ** hostApi,
         pulseaudioHostApi = NULL;
     }
 
-    PaPulseAudio_UnLock( pulseaudioHostApi->mainloop );
     return result;
 }
 
-- 
2.39.3 (Apple Git-145)

